### ISSUE TITLE
reportUndefinedVariable is a sub category of reportGeneralTypingIssues.

### ISSUE THREAD

**Issue by Gobot1234** [microsoft/pyright#3229]
**Describe the bug**
After updating some of my code with unsafe casts to the type checker with the new support for `# type: ignore` comments on a per-line basis #3143, I noticed that on lines with NameErrors reportGeneralTypingIssues was supressing any NameErrors I had on the line.

**To Reproduce**
```py
NameErrorPlease
NameErrorPlease  # type: ignore
NameErrorPlease  # type: ignore[reportGeneralTypingIssues]
NameErrorPlease  # type: ignore[reportUndefinedVariable]
```
```sh
$ pyright test.py        
No configuration file found.
pyproject.toml file found at /Users/James/PycharmProjects/steam.py.
Loading pyproject.toml file at /Users/James/PycharmProjects/steam.py/pyproject.toml
Assuming Python version 3.10
Assuming Python platform Darwin
Auto-excluding **/node_modules
Auto-excluding **/__pycache__
Auto-excluding **/.*
venvPath not specified, so venv settings will be ignored.
stubPath /Users/James/PycharmProjects/steam.py/typings is not a valid directory.
Searching for source files
Found 1 source file
/Users/James/PycharmProjects/steam.py/test.py
  /Users/James/PycharmProjects/steam.py/test.py:1:1 - error: "NameErrorPlease" is not defined (reportUndefinedVariable)
1 error, 0 warnings, 0 informations 
Completed in 2.92sec
```

**Expected behavior**
The first `# type: ignore` comment should supress the `NameError`, the second should not and the third should.

**VS Code extension or command-line**
pyright --version
pyright 1.1.231

**Comment by erictraut** [comment]
Yes, this is correct. The `# type: ignore` is defined by PEP 484 and doesn't honor any rule-specific overrides. You can add anything after the `# type: ignore` comment, but pyright will ignore it.

If you want to suppress particular (pyright-specific) diagnostic rules, you need to use `# pyright: ignore[rule-name]`.

**Comment by Gobot1234** [comment]
My issue is only about the specific rule codes/names, I'd like it to still notify me there is a `NameError` (`reportUndefinedVariable`) even if I add `reportGeneralTypingIssues` as the ignore code, the two don't seem related, `NameError`s aren't, in my opinion at least, typing errors.

**Comment by erictraut** [comment]
The check for an undefined variable is reported as part of the `reportGeneralTypingIssues` rule. This is the intended behavior.

We have on occasion separated out some checks into their own diagnostic rule, but we do this only if there's a strong justification — accompanied by feedback from many users. I don't think there's a good justification for this one. We might reconsider in the future if we get sufficient feedback from other users.

**Comment by erictraut** [comment]
If you want to ignore only `reportGeneralTypingIssues ` on that line, you can use:
`# pyright: ignore[reportGeneralTypingIssues]`. In your code sample above, you are incorrectly using `# type: ignore`, which does not support per-rule filtering.

**Comment by Gobot1234** [comment]
> If you want to ignore only `reportGeneralTypingIssues ` on that line, you can use: `# pyright: ignore[reportGeneralTypingIssues]`. In your code sample above, you are incorrectly using `# type: ignore[reportGeneralTypingIssues]`, which does not support per-rule filtering. So `# type: ignore[reportGeneralTypingIssues]` is the same as `# type: ignore`, which suppresses all diagnostics on that line.

Ah right, that makes sense, I see there isn't much you can do short of adding a lot more error codes which is fair enough if you don't want to add them, thank you.
